src/future/future/flatten.rs

```diff
@@ -0,0 +1,46 @@
+use futures_core::ready;
+use std::pin::Pin;
+
+use crate::future::Future;
+use crate::future::IntoFuture;
+use crate::task::{Context, Poll};
+
+#[doc(hidden)]
```

Comment:
Why is this type hidden from the documentation?

Reply:
If https://github.com/async-rs/async-std/pull/443#discussion_r341852759 is addressed, `#[doc(hidden)]` will no longer be in effect either, and can be removed.